x-pack/platform/plugins/shared/alerting/server/routes/alert_delete/apis/last_run/get_alert_delete_last_run_route.test.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { httpServiceMock } from '@kbn/core/server/mocks';
+import { licenseStateMock } from '../../../../lib/license_state.mock';
+import { alertDeleteLastRunRoute } from './get_alert_delete_last_run_route';
+import { mockHandlerArguments } from '../../../_mock_handler_arguments';
+import { alertDeletionClientMock } from '../../../../alert_deletion/alert_deletion_client.mock';
+import { rulesClientMock } from '../../../../rules_client.mock';
+
+jest.mock('../../../../lib/license_api_access', () => ({
+  verifyApiAccess: jest.fn(),
+}));
+
+jest.mock('../../../lib/track_legacy_terminology', () => ({
+  trackLegacyTerminology: jest.fn(),
+}));
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+const alertDeletionClient = alertDeletionClientMock.create();
+const rulesClient = rulesClientMock.create();
+
+describe('alertDeletePreviewRoute', () => {
```

Comment:
Maybe you could add an error test where we expect `"Error getting last run date: ..."` for when `esClient.search` fails in `x-pack/platform/plugins/shared/alerting/server/alert_deletion/lib/get_last_run.ts`.

Reply:
Shouldn't this kind of errors be covered/tested by the framework already?